### ISSUE TITLE
Is it possible to run interactively with the Python API? (line-by-line)

### ISSUE THREAD

**Issue by nikcleju** [CauldronDevelopmentLLC/CAMotics#392]
Using the Python API, is it possible to run a simulation step-by-step? 

Something like this:

- Open a project file
- Run the first G-Code line, pause, inspect the surface
- Run another G-Code line, pause, inspect the surface
- ...

**Comment by jcoffland** [comment]
Yes, but it would be slow.  You can also control the simulation but time.  So you could step forward one second at a time then regenerate the surface and inspect it.  I don't have a full example of this but here's a start: https://github.com/CauldronDevelopmentLLC/CAMotics/blob/master/examples/python/camotics_python_example.py
